core/include/detray/core/detector_metadata.hpp

```diff
@@ -0,0 +1,252 @@
+    template <typename bin_entry_t, typename container_t>
+    using disc_sf_grid =
+        regular_surface_grid_t<ring2D<>::axes<>, bin_entry_t, container_t>;
+
+    // Irregular binning (hopefully not needed)
+
+    // cylindrical grid for the barrel layers
+    /*template <typename bin_entry_t, typename container_t>
+    using irr_cylinder_sf_grid =
+        surface_grid_t<cylinder2D<>::axes<n_axis::bounds::e_closed,
+    n_axis::irregular, n_axis::irregular>, bin_entry_t, container_t>;
+
+    // disc grid for the endcap layers
+    template <typename bin_entry_t, typename container_t>
+    using irr_disc_sf_grid =
+    surface_grid_t<ring2D<>::axes<n_axis::bounds::e_closed, n_axis::irregular,
+    n_axis::irregular>, bin_entry_t, container_t>;
+
+    // 3D cylindrical grid for the entire barrel
+    template <typename bin_entry_t, typename container_t>
+    using irr_cylinder3D_sf_grid =
+        surface_grid_t<cylinder3D<>::axes<n_axis::bounds::e_closed,
+    n_axis::irregular, n_axis::irregular, n_axis::irregular>, bin_entry_t,
+    container_t>;*/
+
+    /// @}
+
+    // @TODO: Switch to inhomogenous b-field and read homogeneous b-field also
+    // from file.
+    using bfield_backend_t =
+        covfie::backend::constant<covfie::vector::vector_d<scalar, 3>,
+                                  covfie::vector::vector_d<scalar, 3>>;
+
+    /// How to index the constituent objects in a volume
+    /// If they share the same index value here, they will be added into the
+    /// same container range without any sorting guarantees
+    enum geo_objects : std::size_t {
+        e_portal = 0,
+        // TODO: Put into grids
+        e_sensitive = 0,  // Sensitive and passive surfaces will be held in the
+        e_passive = 0,    // same acceleration data structure.
+        e_size = 2,       // Every volume holds two acceleration data structures
+        e_all = e_size,   // One for portals and one for all other surfaces.
+    };
+
+    /// How a volume finds its constituent objects in the detector containers
+    /// In this case: One range for sensitive/passive surfaces, one for portals
+    using object_link_type = dmulti_index<dindex_range, geo_objects::e_size>;
+
+    /// How to store and link transforms
+    template <template <typename...> class vector_t = dvector>
+    using transform_store = single_store<__plugin::transform3<detray::scalar>,
+                                         vector_t, geometry_context>;
+
+    /// Give your mask types a name (needs to be consecutive to be matched
+    /// to a type!)
+    enum class mask_ids {
+        e_rectangle2 = 0,
+        e_portal_rectangle2 = 0,
+        e_trapezoid2 = 1,
+        e_annulus2 = 2,
+        e_cylinder2 = 3,
+        e_portal_cylinder2 = 4,
+        e_ring2 = 5,
+        e_portal_ring2 = 5,
+        e_straw_wire = 6,
+        e_cell_wire = 7,
+        e_single1 = 8,
+        e_single2 = 9,
+        e_single3 = 10,
+        e_unbounded_rectangle2 = 11,
```

Comment:
I am not sure if we need to include single (which I don't know how to use) and unbound surface here, but OK